### DISCUSSION TITLE
Isn't bat a replacement for less, more than it is for cat?

### DISCUSSION THREAD

**Discussion by unfa** [sharkdp/bat#2283] [Q&A]
### Question

I've learned about bat and I love it! One thing that bothers me is the tagline and that `bat` is described as a clone of `cat`.

Surely I often use `cat` to print out a file and take a look at it, but I also use cat often to pipe a file through a chain of processes. Surely I could use `dd`, but then I have to type `dd if=FILE |` rather than `cat FILE |`.

So my usage of cat is not just to read files in the terminal.

I have also heard that the original use case for cat (and where the name comes from) was to **conCATenate** files, not to print them out and read. So doing `cat *.xyz > concatenated.xyz` would splice all the files together into one.

Bat seems to be primarly meant for reading in the console and it behaves more like `less` than `cat` in that regard.

`more` was intended for that purpose and `less` is a better version of it.

Usage of `cat` has changed and it is most often used to print out files than to concatenate them, but `bat` also adds extra characters to make the reading easier (adds a header) and features pagination (like `less`).

Wouldn't all that make `bat` more of a replacement for `less`?

### Self-Answer

I have done some testing and found out that `bat` smartly disable all formatting and extra features when piping it's output to another program, making it indeed a great replacement for `cat`!

**Comment by keith-hall** [comment]
It's perhaps worth nothing that `bat` uses your default pager i.e. `less` under the hood, so it can't exactly be a "replacement" for the tool/pager itself (as in `less` still needs to be available on your machine if you want it's features), just potentially a replacement for what you choose to execute/pipe into directly.

**Reply by unfa** [reply]
Oh, that's interesting. That breaks my idea :D

**Reply by keith-hall** [reply]
There is a proposal for `bat` to include it's own pager, which you may be interested in: https://github.com/sharkdp/bat/issues/1053
